lib/daru/io/io.rb

```diff
@@ -202,23 +200,16 @@ def from_csv_prepare_opts opts
 
       def from_csv_hash_with_headers(path, opts)
         opts[:header_converters] ||= :symbol
-
         ::CSV
-          .read(path, 'rb',opts)
-          .tap { |c| yield c if block_given? }
+          .parse(open(path), opts)
           .by_col.map { |col_name, values| [col_name, values] }.to_h
       end
 
       def from_csv_hash(path, opts)
         csv_as_arrays =
-          ::CSV
-          .open(path, 'rb', opts)
-          .tap { |c| yield c if block_given? }
```

Comment:
As the `from_html` module uses a similar logic to differentiate between local and url files, I was thinking about this flow - 

```
def from_html path
    if File.exists? path
        # Parse from File.read 
    else
        begin
            # Parse from URI.parse()
        rescue
            # File not found - raise error / display error message
        end
    end
end
```

(The `if-else` can be converted to `? :` or other forms, if required.)
Do comment your insights on this, @anshuman23 @zverok @v0dro :smile: